sbgECom/src/IO/Logger.cpp

```diff
@@ -0,0 +1,71 @@
+#include "Logger.h"
+#include "rocketState.h"
+
+#include <iostream>
+#include <thread>
+#include <iostream>
+#include <fstream>
+#include <queue>
+
+
+void Logger::initialize() {
+
+}
+
+void Logger::run() {
+    while (true) {
+
+        //timing stuff here
+        //pause the thread
+        dequeueToFile("Log.txt");
+    }
+}
+
+void Logger::enqueueSensorData(rocketState curSensorData) {
+    std::lock_guard<std::mutex> lockGuard(mutex);
+	logQueue.push(curSensorData);
+}
+
+
+void Logger::dequeueToFile(std::string filename) {
+	if (!logQueue.empty()) {
+		rocketState currentState;
+        {
+            std::lock_guard<std::mutex> lockGuard(mutex);
+            currentState = logQueue.front();
+            logQueue.pop();
+        }
+
+		std::ofstream myfile(filename.c_str(), std::ios_base::app); //append
+    	if(myfile.is_open()) {
+			myfile << "Xangle: "<< currentState.sbg.Xangle << "\n";
+			myfile << "Yangle: "<< currentState.sbg.Yangle << "\n";
+			myfile << "Zangle: "<< currentState.sbg.Zangle << "\n";
+
```

Comment:
Looks like the indentation got messed up here. We should run it through a formatter